Improve performance

ehmicky opened the pull request:
This PR improves performance by ~40% (on my machine). The performance improvement seems fairly consistent with both very small and very objects.

Benchmark code:

```js
import filterObj from 'filter-obj'

const bigObject = Object.fromEntries(new Array(1e4).fill().map((_, index) => [`a${index}`, index]))
const predicate = (key, value) => value % 2 === 0

console.time()
for (let i = 0; i < 1e3; i += 1) {
	filterObj(bigObject, predicate)
}
console.timeEnd()
```

The performance improvement is due to avoiding to create an unnecessary array with `Object.entries()`.

I have also opened #13 to ensure this does not change any behavior.

sindresorhus commented on index.js:
```diff
@@ -2,16 +2,21 @@ export default function filterObject(object, predicate) {
-				enumerable: true,
-				configurable: true,
-			});
+	for (const key in object) {
```
You need to add a short comment on why this is not using `Object.entries` or `Object.keys` to prevent this being reverted in the future.

ehmicky replied:
Done :+1:

sindresorhus commented:
What Node.js version? It should be tested on Node.js 18.

sindresorhus commented:
Can you commit the benchmark code too?

sindresorhus commented:
> The performance improvement is due to avoiding to create an unnecessary array with Object.entries().

How about `Object.keys()`?

ehmicky commented:
I have added `benchmark.js`.
The results above were with Node 18.

I just tried with `Object.keys()` and it is actually much faster than `Object.entries()`. Its performance is close to `for ... in`. On my machine, `for ... in` is faster when the object has roughly less than 100 keys, but `Object.keys()` is faster when it has roughly more than 100 keys. In other words, `for ... in` would be faster for the common case (small-medium objects) but `Object.keys()` would be faster for the worst case (big objects). Also, `Object.keys()` leads to slightly simpler code.

Based on this, I changed the logic to using `Object.keys()` instead.
